alicloud/table_alicloud_cs_kubernetes_cluster.go

```diff
@@ -448,8 +461,44 @@ func getCsKubernetesClusterLog(ctx context.Context, d *plugin.QueryData, h *plug
 	return nil, nil
 }
 
-func getCsKubernetesClusterAka(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
-	plugin.Logger(ctx).Trace("getCsKubernetesClusterAka")
+func getCsKubernetesClusterNamespace(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	region := plugin.GetMatrixItem(ctx)[matrixKeyRegion].(string)
+	plugin.Logger(ctx).Trace("getCsKubernetesClusterNamespace")
+
+	var id string
+	if h.Item != nil {
+		clusterData := h.Item.(map[string]interface{})
+		id = clusterData["cluster_id"].(string)
+	} else {
+		id = d.KeyColumnQuals["cluster_id"].GetStringValue()
+	}
+
+	client, err := ContainerService(ctx, d, region)
+	if err != nil {
+		return nil, nil
+	}
+
+	request := requests.NewCommonRequest()
+	request.Method = "GET"
+	request.Scheme = "https"
+	request.Domain = "cs.aliyuncs.com"
+	request.Version = "2015-12-15"
+	request.PathPattern = "/k8s/" + id + "/namespaces"
+	request.Headers["Content-Type"] = "application/json"
+	request.QueryParams["RegionId"] = region
+	body := `{}`
+	request.Content = []byte(body)
```

Comment:
Do we require to define all this parameters?
Can't we use default values?

Reply:
Updated the params with required data only